pyproject.toml

```diff
@@ -16,6 +16,7 @@ dependencies = [
     "numpy",
     "torch>=2.6.0",
     "cached-path>=1.7.2",
+    "cloudpathlib[gs]",
```

Comment:
We have functions in `olmo_core.io` that implement what you need already, so I'd rather use those instead of adding a new dependency, especially when that dependency is only used in internal scripts.